src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -2424,196 +2582,233 @@ void GraphOptimizer::MergeTransposeAndReorder(Graph &graph) {
-        // inPlace memory is not used
-        const auto parentInPlace = parentNode->getParentEdgeAt(0)->inPlace(Edge::LOOK_UP);
-        const auto& childEdges = childNode->getChildEdgesAtPort(0);
-        const auto childInPlace = std::any_of(childEdges.begin(), childEdges.end(),
-            [](const EdgePtr& edge){ return edge->inPlace(Edge::LOOK_DOWN); });
+        auto transposeOrder = updateOrder(transposeNode, reshapeNode);
+        auto descBeforeReorder = reorderNode->getParentEdgesAtPort(0)[0]->getParent()->getSelectedPrimitiveDescriptor()->getConfig().outConfs[0].getMemDesc();
+        auto layoutOrder = descBeforeReorder->as<BlockedMemoryDesc>()->getOrder();
 
-        bool isOptimized = !(parentInPlace && childInPlace);
-
-        graph.DropNode(parentNode);
-        graph.DropNode(childNode);
+        auto inBlockedDesc = reorderNode->getSelectedPrimitiveDescriptor()->getConfig().inConfs[0].getMemDesc()->as<BlockedMemoryDesc>();
+        auto outBlockedDesc = reorderNode->getSelectedPrimitiveDescriptor()->getConfig().outConfs[0].getMemDesc()->as<BlockedMemoryDesc>();
 
-        auto inDesc = parentNode->getSelectedPrimitiveDescriptor()->getConfig().inConfs[0].getMemDesc();
-        auto outDesc = childNode->getSelectedPrimitiveDescriptor()->getConfig().outConfs[0].getMemDesc();
+        auto& inOrder = inBlockedDesc->getOrder();
+        auto& outOrder = outBlockedDesc->getOrder();
 
-        auto inPrec = inDesc->getPrecision();
-        auto outPrec = outDesc->getPrecision();
+        if (checkAscendingSummaryOrder(transposeOrder, layoutOrder, inOrder, outOrder)) {
+            mergeTransposeReshapeReorder(graph, transposeNode, reshapeNode, reorderNode, false);
+        }
+    }
+}
 
-        auto reorderInDesc = inDesc;
-        auto reorderOutDesc = outDesc->cloneWithNewPrecision(inPrec);
+void GraphOptimizer::MergeReorderAndTranspose(Graph &graph) {
+    auto& graphNodes = graph.GetNodes();
 
-        std::string reorderlayerName = parentParentNode->getName() + "_" +
-                Reorder::getReorderArgs(*reorderInDesc, *reorderOutDesc) + "_" + "fake";
+    auto isSuitableTranspose = [](NodePtr node) {
+        return node->getType() == Type::Transpose
+                && node->getChildEdges().size() == 1
+                && !node->isDynamicNode();
+    };
 
-        DEBUG_LOG("mergeTransposeAndReorder ", parentNode->getName(), " and ", childNode->getName(), " -> ", reorderlayerName);
+    auto isSuitableReshape = [](NodePtr node) {
+        if (node->getChildEdges().size() != 1)
+            return false;
+        // Reshape supported only in one case: if two consecutive input dims are merged into 1
+        const auto& inShape = node->getInputShapeAtPort(0).toPartialShape();
+        const auto& outShape = node->getOutputShapeAtPort(0).toPartialShape();
+        if (inShape.size() - outShape.size() != 1)
+            return false;
```

Comment:
Would you please elaborate this limitation? In my mind, Reshape operator only manipulates shape, and keeps memory data unchanged. So there should be no restriction to it here.

Reply:
We have this limitation for reshape because of the `updateOrder` logic. Please see [the comment with an example](https://github.com/openvinotoolkit/openvino/pull/18999/files/fcea9306098ac506f318550a39cd3c1b1fba0026#diff-4596e5068af72535c11480ba99ca338f9097ce55ea043d33a46cddff28c81aedR2738) (it is like a swap of Reshape and Transpose in a graph). This swap-like logic can't be applied to an arbitrary Reshape - so we have to have such limitations (The same is true for the reverse operations order)